game/features/auto_splitter.cpp

```diff
@@ -0,0 +1,247 @@
+#include "auto_splitter.h"
+
+#include "game/kernel/jak1/kmachine.h"
+#include <game/kernel/common/Ptr.h>
+#include <game/kernel/common/kmachine.h>
+#include <game/kernel/jak1/kscheme.h>
+
+#include "third-party/fmt/core.h"
+
+AutoSplitterBlockJak1 gJak1AutoSplitterBlock;
+
+void update_autosplitter_block_jak1(u32 jak1_autosplit_info) {
+  auto info = jak1_autosplit_info ? Ptr<jak1::AutoSplitInfoJak1>(jak1_autosplit_info).c() : NULL;
```

Comment:
I think this check is redundant - doing `Ptr<Whatever>(0).c()` will give you `NULL`:

```
  /*!
   * Convert to a C pointer.
   */
  T* c() {
    if (!offset) {
      return nullptr;
    }
    return (T*)(g_ee_main_mem + offset);
  }
  ```